compiler/src/iree/compiler/Codegen/LLVMCPU/KernelDispatch.cpp

```diff
@@ -1913,11 +2014,17 @@ getMmt4dLoweringConfig(linalg::LinalgOp op) {
   vecTileSizes[mmt4dDimBase + 3] = M0;
   vecTileSizes[mmt4dDimBase + 4] = N0;
   vecTileSizes[mmt4dDimBase + 5] = K0;
-  limitVectorTileSizes(op, vecTileSizes);
-
+  IREE::Codegen::ScalableTileFlags vecScalableTileFlags(mmt4dDimBase + 6,
+                                                        false);
+  if (!(targetConfig && isAArch64(targetConfig) &&
+        isScalableVectorizationEnabled() &&
+        succeeded(adjustVectorSizesForScalableVectorization(
+            op, targetConfig, M0, N0, vecTileSizes, vecScalableTileFlags)))) {
+    limitVectorTileSizes(op, vecTileSizes);
+  }
```

Comment:
This is too hard to follow.. Can we just create different functions for scalable and non-scalable cases for getting the vector sizes?